tests/framework/e2e/etcdctl.go

```diff
@@ -507,3 +508,60 @@ func (ctl *EtcdctlV3) UserChangePass(user, newPass string) error {
 	_, err = cmd.Expect("Password updated")
 	return err
 }
+
+func (ctl *EtcdctlV3) Watch(ctx context.Context, key string, opts config.WatchOptions) clientv3.WatchChan {
+	args := ctl.cmdArgs()
+	args = append(args, "watch", key)
+	if opts.RangeEnd != "" {
+		args = append(args, opts.RangeEnd)
+	}
+	args = append(args, "-w", "json")
+	if opts.Prefix {
+		args = append(args, "--prefix")
+	}
+	if opts.Interactive {
+		args = append(args, "--interactive")
+	}
+	if opts.Revision != 0 {
+		args = append(args, "--rev", fmt.Sprint(opts.Revision))
+	}
+	if len(opts.ExecCmd) > 0 {
+		args = append(args, "--")
+		args = append(args, opts.ExecCmd...)
+	}
+
+	proc, err := SpawnCmd(args, nil)
+	if err != nil {
+		return nil
+	}
+	defer proc.Stop()
+	if opts.Interactive {
+		wl := strings.Join(args, " ") + "\r"
+		if err = proc.Send(wl); err != nil {
+			return nil
+		}
+	}
+
+	ch := make(chan clientv3.WatchResponse)
+	go func() {
+		var resp clientv3.WatchResponse
+		for {
+			lines := proc.Lines()
+			for _, line := range lines {
+				err = json.Unmarshal([]byte(line), &resp)
+				if err != nil {
+					continue
+				}
+				ch <- resp
+			}
+
+			select {
+			case <-ctx.Done():
```

Comment:
Won't this wait always until `ctx.Done()`? I think we should default case in select so the external for loop makes sense, however I don't internal loop will look as `proc.Lines()` will repeat lines.